iceoryx_posh/source/mepoo/mem_pool.cpp

```diff
@@ -39,7 +39,7 @@ MemPoolInfo::MemPoolInfo(const uint32_t usedChunks,
 
 constexpr uint64_t MemPool::CHUNK_MEMORY_ALIGNMENT;
 
-MemPool::MemPool(const greater_or_equal<uint32_t, CHUNK_MEMORY_ALIGNMENT> chunkSize,
+MemPool::MemPool(const greater_or_equal<uint64_t, CHUNK_MEMORY_ALIGNMENT> chunkSize,
```

Comment:
Here we also need to check for potential overflows when `chunkSize` is multiplied with `numberOfChunks`

Reply:
@elBoberido yes, but I would call here again panic and not work with an expected.

Reply:
I'd also like to see an explicit IOX_EXPECTS verifying that `chunkSize` * `numChunks` isn't somehow greater than `numeric_limits<uint64_t>::max`. Such a check would have helped previously when I discovered the `numeric_limits<uint64_t>::max` overflow bug.

Reply:
Well, `chunkSize * numChunks` will never be larger than `numeric_limits<uint64_t>::max` ;)

We have to check whether `numeric_limits<uint64_t>::max / numChunks` is larger or equal to `chunkSize`.

Side note, it is now called `IOX_ENFORCE` :)

I would avoid to panic here, an incorrect size is not different from an incorrect alignment, which also does not panic. Let's keep doing it like the local error handling on the class instead of mixing it. It makes testing also a little bit easier.

Reply:
Let me rephrase - I'd like a check that the multiplication will not overflow. There is pseudocode for this in this stackoverflow: https://stackoverflow.com/a/1815391

Reply:
And it looks like the check below does indeed implement the check correctly 👍